### PR TITLE
Added Claimant Model for Maps, discussing CSMs

### PR DESCRIPTION
**Pull request by mhutchinson**
This is being increasingly used as foundations for proposals in the public domain, so now is a good time to get some content made public which outlines these themes.

This also tidies up some unclosed tags in the logs docs

### CODE REVIEW

File: docs/claimantmodel/Maps.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+This doc presents a model for transparent ecosystems built around maps. This doc requires understanding the core [Claimant Model](CoreModel.md), and it is recommended that [Claimant Model: Logs](Logs.md) is also read first.
+
+## Groundwork: Discoverability
+
+A Claim is discoverable if it can be found by the Claim Verifier without information being passed to them by the Believer.
+The Log Claimant Model states that the purpose of Log Transparency is to enable discoverability of Claims.
+To understand whether this statement applies to Maps, we need to understand what it is about Logs that enables discoverability.
+
+A Log can be thought of as a Map where the key is the leaf index, and new entries need to be written into the next available leaf.
```
Comment (AlCutter) [inline docs/claimantmodel/Maps.md:11]: I think it's worth introducing the key concept of a key (:D) somewhere above this line - the idea of a key in a log is a bit strange without context, but maybe if we just casually use the phrase "key value map" somewhere (maybe in the first sentence?) I think it'll cue this up nicely.

File: docs/claimantmodel/Maps.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+Maps and Logs both have a single root hash which precisely commits to key/value data, and given this root hash and a key, there is only one verifiable value for the point lookup.
+The key difference (pun intended) is that in Maps the writer gets to choose any arbitrary key, where in Logs the key is the next available sequence number.
+
+A Log Checkpoint commits to a range of keys by stating the tree size.
+Tree size is the only input required to generate all keys for a log, and it is this property of key enumerability that makes Claim discoverability simple; given a Log Checkpoint, a Verifier can discover all Claims by requesting all entries from [0..treeSize).
+Thus providing the Verifier is getting recent Checkpoints and is seeing Checkpoints consistent with the Believer, each and every Claim will be discovered in time.
```
Comment (AlCutter) [inline docs/claimantmodel/Maps.md:17]: "Thus, providing"

File: docs/claimantmodel/Maps.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+While simple, this is not efficient, especially for Verifiers interested in a tiny subset of Claims.
+
+Map keys cannot be generated/discovered from a Map Checkpoint, thus Maps do not allow for efficient discoverability through key enumeration.
```
Comment (AlCutter) [inline docs/claimantmodel/Maps.md:20]: thus -> consequently for variety

File: docs/claimantmodel/Maps.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+CSMs allow for more efficient discoverability of Claims, particularly in ecosystems where there are many independent Claim Verifiers with distinct Claim Subjects.
+Claim Verifiers can become somewhat ignorant and selfish, looking after only their own Subjects and trusting others to look after their own.
+The trade-off for this is the risk that any Claims placed in the map at locations the Verifier does not expect will not be discovered by the Verifier.
+This risk can be eliminated with careful design of Claim Subjects and how they are extracted from Claims.
+
+### Requirements
+
+To use a CSM for discoverability, these properties must hold:
+ 1. Claim Subjects can be derived from only the Claim. Examples (Claim Subjects in bold):
+    * CT: the Claim (cert) contains the **domain(s)** the certificate is for
```
Comment (AlCutter) [inline docs/claimantmodel/Maps.md:49]: I'd be tempted to expand to Certificate Transparency.

Also, "golang" (or maybe "Golang")?

File: docs/claimantmodel/Maps.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+This example also highlights that mapFn should not return too many Subjects. While the empty domain `.` may be derived from all Claims, doing so entails a map key which is effectively committing to all entries that would have been in the log. If this is the only mechanism that permits discoverability then a log should have been chosen instead of a map.
+
+### CSM in the Claimant Model
+#### Log-Backed Maps & Mogs
+
+If the map is constructed from a Claim Log then we have the following:
+
+<dl>
+<dt>Claim<sup>CSM</sup></dt>
+<dd><i>"Applying $mapFn to the logged data in $LogCheckpoint results in the map with $rootHash"</i></dd>
```
Comment (AlCutter) [inline docs/claimantmodel/Maps.md:90]: /in/committed to by/

File: docs/claimantmodel/Maps.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+<dt>Statement<sup>CSM</sup></dt>
+<dd>Map Checkpoint (sometimes SMR, or Signed Map Root)</dd>
```
Comment (AlCutter) [inline docs/claimantmodel/Maps.md:92]: which itself contains/commits to at least $LogCheckpoint.Size

File: docs/claimantmodel/Maps.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+<dd>A Verifier<sup>LOG,DOMAIN</sup> with the power to compute the map</dd>
+<dt>Arbiter<sup>CSM</sup></dt>
+<dd>CSM Arbiter, possibly the same as Arbiter<sup>LOG</sup></dd>
+</dl>
+
+#### General Maps & Mogs
+If the map is not constructed from a Claim Log, then the Claim is a lot more nuanced.
+The Map Checkpoint needs to explicitly include a Revision identifier that has a total ordering.
+
+General Map Claim:
```
Comment (AlCutter) [inline docs/claimantmodel/Maps.md:107]: It feels like this is an "Insert-only Map Claim"?
I think we came to the conclusion that in the absence of a log of claimed deltas the General Map claim is probably "I claim this is a map root hash <shrug>", and if it has revisions (either implicitly = there _is_ only one and it's static, or explicitly with rev numbers) then "there's only 1 root hash at revision x"?

**Review by AlCutter — APPROVED** [review]
Few minor nits/suggestions/thoughts

File: docs/claimantmodel/Maps.md (hunk 6 above)
Comment (mhutchinson) [reply docs/claimantmodel/Maps.md:93]: I'm going to merge this as currently written. Happy to continue this discussion and make changes in future PRs.

**Pull request merged**